src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,138 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_SIZE = 10;
+    private static final int ADDITIONAL_SIZE = 5;
+    private static final int NOT_FOUND = -1;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[START_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (!isFullArray()) {
+            array[size] = value;
+            size++;
+        } else {
+            resizeArray();
+            add(value);
```

Comment:
Try to simplify this only invoke method resize and implement check and other logic inside